site-src/geps/gep-1619.md

```diff
@@ -0,0 +1,89 @@
+
+## Goals
+- Define session persistence and session affinity to establish a common language
+- Identify differences in session persistence functionality between implementations
+
+## Non-Goals
+- Create proposals for adding session persistence configuration to Gateway API object specification
+- Mandate a default or expected session persistence functionality for implementations
+- Require implementations to support session persistence or a specific form of it
+
+## Introduction
+
+At the November 28th, 2022 SIG Gateway API meeting, we discussed the idea of standardizing and documenting session persistence (also known as session stickiness). It was recommended that we first generate discussion for how we describe session persistence, agree on how it is achieved, and find out which implementers support session persistence. 
+
+Conformance test for session persistence will be considered if deemed appropriate. Furthermore, this information can be used for developing a Gateway API spec to configure session persistence in the future.
+
+### Defining Session Persistence
+
+Session persistence is when a client request is directed to the same backend server for the duration of a "session". It is achieved when a client directly provides information, such as a header, that a proxy uses as a reference to direct traffic to a specific server. Persistence is an exception to load balancing: a persistent client request bypasses the proxy's load balancing algorithm, going directly to a backend server it has previously established a session with.
+
+Some of the concerns of session persistence are the duration and expiration of the session, security of the transaction stream, and storage of the context or state. Application developers use session persistence to improve the performance and reduce overall storage needs by aligning each transaction and its cached data with the same server.
+
+Session affinity, not to be confused with session persistence, uses an existing attribute of the request to consistently send to the same backend. Session affinity can be considered a weaker form of session persistence: it is not guaranteed to persist a connection to the same backend server if certain attributes of the request or the backends are changed.
+
+### Achieving Session Persistence and Session Affinity
+
+Session persistence is achieved with the application layer, while session affinity often uses, but is not limited to, attributes below the application layer.
+
+Here are a some mechanisms to achieving both:
+
+**Cookie-Based Session Persistence**
+
+The most common mechanism is by using cookies (described by [RFC6265](https://www.rfc-editor.org/rfc/rfc6265)) with the set-cookie HTTP response header. A client will use the provided value in the set-cookie response header in a cookie request header in subsequent requests. Proxies can use this cookie header to maintain a persistent connection to a single backend server on behalf of the client.
```

Comment:
In the next section you have "backend providing an arbitrary name". This section doesn't specify who provides the cookie. 

I think in both cases we need to make it a bit more clear if we want to be compatible with broadly used patterns.

1. A backend may initiate the session as needed ( Java getHttpSession(), etc) - the Gateway API should allow this and not force but allow a persistent session. In this case - the contract needs to be very clear and we should mention that the gateway implementation may alter the cookie ( to add backend info ) or have ability to decode part of the cookie (servlets and many others use a suffix to identify the node).
2. The gateway implementation itself may generate the header/cookie - in this case backend no longer has a choice to use or not sessions. 

In Java spec, a lot of discussion was around on the 'shopping cart' use case - a user may browse the store searching ( using proper load balancing ), but when he adds to cart the persistent session is created. The other important use case was auth/SSO - a page may be accessed as unauthenticated, but once user logs in usually
a cookie is used and in most implementations it creates a session. 

Having persistent sessions tied to the user is still broadly used.